src/firmware/app/primitives/move_helper.c

```diff
@@ -18,125 +18,32 @@
-{
-    (*wheel_axes)[0] = angle + ANGLE_TO_FRONT_WHEELS - PI_2;
-    (*wheel_axes)[1] = angle + ANGLE_TO_FRONT_WHEELS + PI_2;
-    (*wheel_axes)[2] = angle - ANGLE_TO_FRONT_WHEELS - PI_2;
-    (*wheel_axes)[3] = angle - ANGLE_TO_FRONT_WHEELS + PI_2;
-    (*wheel_axes)[4] = angle + ANGLE_TO_BACK_WHEELS - PI_2;
-    (*wheel_axes)[5] = angle + ANGLE_TO_BACK_WHEELS - (3 * PI_2);
-    (*wheel_axes)[6] = angle - ANGLE_TO_BACK_WHEELS + PI_2;
-    (*wheel_axes)[7] = angle - ANGLE_TO_BACK_WHEELS + (3 * PI_2);
-}
-
-unsigned choose_wheel_axis(float dx, float dy, float current_angle, float final_angle)
-{
-    float wheel_axes[8];
-    build_wheel_axes(&wheel_axes, current_angle);
-    // the angle on the global axis corresponding to the bot's movement
-    float theta_norm = atan2f(dy, dx);
-    // initialize a variable to store the minimum rotation
-    float minimum_rotation = 2 * P_PI;
-    // the index that corresponds to the minimum rotation
-    unsigned min_index = 0;
-    unsigned i;
-    // loop through each axis to find the optimal one to rotate onto.
-    // it should be the axis that is closest to our final angle
-    for (i = 0; i < 2 * NUMBER_OF_WHEELS; i++)
-    {
-        float relative_angle_to_movement = min_angle_delta(wheel_axes[i], theta_norm);
-        float initial_rotation           = current_angle + relative_angle_to_movement;
-        float abs_final_rotation = fabsf(min_angle_delta(initial_rotation, final_angle));
-        // if we have found a smaller angle, then update the minimum rotation
-        // and chosen index
-        if (abs_final_rotation < minimum_rotation)
-        {
-            minimum_rotation = abs_final_rotation;
-            min_index        = i;
-        }
-    }
-    return min_index;
-}
-
-/**
- * If we are far enough away from our destination, then we should try
- * rotating onto a wheel axis so that we can move faster. We should
- * pick the wheel axis that minimizes the distance the bot will have
- * to rotate to get to its destination angle after rotating onto an
- * axis.
- *
- * @param optimal_wheel_axes_index The index of the wheel axes to rotate onto
- * @param pb [in/out] The data container that contains information about
- * the direction the robot will move along.
- * @param angle The angle that the robot is currently facing
- */
-void choose_rotation_destination(unsigned optimal_wheel_axes_index, PhysBot* pb,
-                                 float angle)
-{
-    // if we are close enough then we should just allow the bot to rotate
-    // onto its destination angle, so skip this if block
-    if (fabsf(pb->maj.disp) > APPROACH_LIMIT)
-    {
-        float wheel_axes[8];
-        build_wheel_axes(&wheel_axes, angle);
-        float theta_norm = atan2f(pb->pos[1], pb->pos[0]);
-        // use the pre-determined wheel axis
-        pb->rot.disp = min_angle_delta(wheel_axes[optimal_wheel_axes_index], theta_norm);
-    }
-}
-
-
 void plan_move_rotation(PhysBot* pb, float avel)
```

Comment:
Function seems unused? It also seems to be planning everything, not just the rotation?

Reply:
The folds on github confused this man, this comment is not relevant :joy: